samples/subsys/mgmt/updatehub/README.rst

```diff
@@ -7,176 +7,341 @@ Overview
+    :shield: atmel_rf2xx_arduino
+    :goals: build
+    :compact:
+
+Step 4.5: Build for BLE IPSP [experimental]
+-------------------------------------------
+
+The BLE IPSP needs ``overlay-ipsp.conf``.  This may requires two nodes:
+one will be the host and the second one will be the device under test.  The
+validation needs a Linux kernel >= 4.9 with all 6loWPAN support.  In this
+particular case the Bluetooth 6LoWPAN module is needed. The start point is try
+reproduce the Zephyr :ref:`bluetooth-ipsp-sample`. It is out of scope
+at this moment provide support since it is experimental.  The gateway was
+tested with native linux driver and an USB dongle.
+
+.. zephyr-app-commands::
+    :zephyr-app: zephyr/samples/subsys/mgmt/updatehub
+    :board: nrf52840dk_nrf52840
+    :build-dir: app
+    :gen-args: -DOVERLAY_CONFIG="overlay-ipsp.conf;overlay-prj.conf"
     :goals: build
+    :compact:
+
+Step 4.6: Build for OpenThread Network [experimental]
+-----------------------------------------------------
+
+The OpenThread requries the ``overlay-ot.conf``.  It requires two nodes:
+one will be the host NCP and the second one will be the device under test.  The
+validation needs a Linux kernel >= 4.9 with optional NAT-64 support.  The
+start point is try reproduce the `OpenThread Router`_. It is
+out of scope at this moment provide support since it is experimental.  The
+gateway was tested using two boards with OpenThread 1.1.1 on NCP mode.
 
-.. _updatehub_sample_sign:
+.. zephyr-app-commands::
+    :zephyr-app: zephyr/samples/subsys/mgmt/updatehub
+    :board: nrf52840dk_nrf52840
+    :build-dir: app
+    :gen-args: -DOVERLAY_CONFIG="overlay-ot.conf;overlay-prj.conf"
+    :goals: build
+    :compact:
 
-Step 5: Sign the first image
-============================
 
-From this section onwards you use a binary (``.bin``) image format.
+Step 5: Sign the app image
+==========================
 
-Using MCUboot's :file:`imgtool.py` script, sign the :file:`zephyr.bin`
-file you built in Step 3. In the below example, the MCUboot repo is located at
-:file:`~/src/mcuboot`.
+The app image is the application itself that will be on the board.  This app
+will connect to UpdateHub server and check for new images.  The image will be
+loaded on the board with version 1.0.0.  It is important check what file
+format you SoC tools uses.  In general, Zephyr can create images with binary
+(``.bin``) image format or Intel's (``.hex``) image format.
+
+The Zephyr provide the ``west`` tool that simplify the signing process.  Just
+call west with proper parameter values:
 
 .. code-block:: console
 
-   ~/src/mcuboot/scripts/imgtool.py sign \
-	    --key ~/src/mcuboot/root-rsa-2048.pem \
-	    --align 8 \
-	    --version 1.0.0 \
-	    --header-size 0x200 \
-	    --slot-size <image-slot-size> \
-	    --pad \
-        <path-to-zephyr.bin> signed.bin
+  west sign -t imgtool -d build/app -- --version 1.0.0 --pad
```

Comment:
Build system might does that step automatically if  `CONFIG_MCUBOOT_SIGNATURE_KEY_FILE` is provided.
See also `CONFIG_MCUBOOT_EXTRA_IMGTOOL_ARGS` and `CONFIG_MCUBOOT_GENERATE_CONFIRMED_IMAGE`

Reply:
My afraid is If we add this options right now it may confuse users. UpdateHub sample right now need instruct users about create two different image versions: 1) it will be used to flash the board, 2) is a different version to package for OTA. Not all parameters are equal for both images and for those boards that only have HEX enabled by default will be a problem when build/sign [2], because OTA needs the package with the BIN signed file not HEX.

My plan is enhance west to package UpdateHub. Than we can keep extra parameters internal for the second image sign and release to user the west flash/package without any confusion or mistake.

Reply:
ok, I just want be sure you are aware of.